lib/much-decimal.rb

```diff
@@ -1,5 +1,48 @@
-require "much-decimal/version"
+require 'much-plugin'
+
+require 'much-decimal/version'
 
 module MuchDecimal
+  include MuchPlugin
+
+  DEFAULT_PRECISION = 2.freeze
+
+  def self.integer_to_decimal(integer, precision)
+    if integer.respond_to?(:to_i) && !integer.to_s.empty?
+      base_10_modifier = (10.0 ** precision)
+      integer.to_i / base_10_modifier
+    end
+  end
+
+  def self.decimal_to_integer(decimal, precision)
+    if decimal.respond_to?(:to_f) && !decimal.to_s.empty?
+      base_10_modifier = (10.0 ** precision)
+      (decimal.to_f * base_10_modifier).round.to_i
+    end
+  end
+
+  plugin_included do
+    extend ClassMethods
+  end
+
+  module ClassMethods
+
+    def decimal_as_integer(attribute, options = nil)
+      options ||= {}
+      source    = options[:source] || "#{attribute}_as_integer"
+      precision = (options[:precision] || DEFAULT_PRECISION).to_i
+
+      define_method(attribute) do
+        integer_as_decimal = self.send(source)
+        MuchDecimal.integer_to_decimal(integer_as_decimal, precision)
+      end
+
+      define_method("#{attribute}=") do |decimal_as_integer|
+        new_value = MuchDecimal.decimal_to_integer(decimal_as_integer, precision)
```

Comment:
@jcredding likewise, can we just call the block arg `decimal` and the local var `integer`?  I'm just trying to keep it obvious the state of the data at each step.